### PR TITLE
fix: preserve reasoning in assistant tool-call conversion

### PR DESCRIPTION
**Pull request by skulldogged**
## Summary

Preserve assistant reasoning content when converting assistant messages that include tool calls. This fixes provider request failures when reasoning-enabled models expect `reasoning_content` on assistant tool-call messages.

## What Changed

**`src/llm/model.rs`**
- Added `ReasoningContent` import from Rig
- Updated `convert_messages_to_openai()` to collect `AssistantContent::Reasoning` text during message conversion
- Updated `convert_messages_to_openai_responses()` with the same reasoning preservation logic
- Added helper function `collect_reasoning_text_parts()` to safely extract reasoning text from various content types
- Included `reasoning_content` on assistant messages when reasoning is present, including in tool-call flows
- Added 4 new unit tests covering reasoning-bearing assistant tool-calls in both conversion paths, plus edge cases with redacted/encrypted reasoning

**`src/tools/send_message_to_another_channel.rs`**
- Minor refactoring: replaced nested if blocks with Rust's let-else pattern for cleaner conditionals

## Why

Reasoning-enabled models (like those using Moonshot AI) emit reasoning content and expect it to be carried through serialization. Without `reasoning_content` in the OpenAI request format, these models fail with errors like: `thinking is enabled but reasoning_content is missing in assistant tool call message at index 4`. This ensures reasoning context is preserved end-to-end, preventing both context loss and outright request failures.

## Testing

Added unit tests in `src/llm/model.rs`:
- `convert_messages_to_openai_preserves_reasoning_for_tool_calls` — verifies reasoning + tool calls are serialized correctly
- `convert_messages_to_openai_responses_preserves_reasoning_for_tool_calls` — same for responses API path
- `convert_messages_to_openai_preserves_empty_reasoning_content_for_redacted_reasoning` — edge case with redacted reasoning
- `convert_messages_to_openai_responses_preserves_empty_reasoning_content_for_redacted_reasoning` — edge case for responses API

> [!NOTE]
> This PR preserves assistant reasoning content in message serialization for both OpenAI request and response API conversion paths. Four new unit tests verify correct handling of reasoning in tool-call messages and edge cases with redacted/encrypted reasoning content.
>
> <sub>Written by [Tembo](https://app.tembo.io) for commit [1e89e96](https://github.com/spacedriveapp/spacebot/commit/1e89e96009e0b0bf5793d24d0d691e82368d7783).</sub>

### CODE REVIEW

File: src/llm/model.rs [hunk 1]
```diff
@@ -1410,6 +1405,24 @@ fn convert_messages_to_openai(messages: &OneOrMany<Message>) -> Vec<serde_json::
...
+            }
+            ReasoningContent::Summary(summary) => {
+                (!summary.trim().is_empty()).then(|| summary.clone())
+            }
+            ReasoningContent::Encrypted(_) | ReasoningContent::Redacted { .. } => None,
+            #[allow(unreachable_patterns)]
+            _ => None,
+        })
+        .collect()
+}
```
[the remark this answers is not in the document]
Comment (skulldogged) [reply src/llm/model.rs:1443]: Thanks — I looked into this, and I agree it points at a broader correctness/design issue for native OpenAI Responses roundtripping. I am intentionally leaving that out of this PR because this change is scoped to fixing the immediate provider failure path we hit through OpenCode Go/Moonshot, where reasoning-enabled assistant tool-call messages were rejected when `reasoning_content` was missing.

This PR keeps the smallest change that fixes that concrete failure and preserves the current request shape used elsewhere. Refactoring the Responses conversion to emit structured `reasoning` input items would be a larger follow-up that should be handled separately so it can be reviewed and validated as its own behavior change.

**Review by jamiepine — APPROVED** [review]
